### DISCUSSION TITLE
Requests limit

### DISCUSSION THREAD

**Discussion by scffs** [community/community#50114] [Apps, API and Webhooks] [answered]
### Select Topic Area

Question

### Body

Hi y'all !
I am developing an application (bot) to work with the GitHub API and began to face a limit on the number of requests. Is there any way to increase it or solve the problem in another way?

```
 message: "API rate limit exceeded for MY_IP. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.)",
```
P.S. I used the GitHub App token

**Comment by Anmol-Baranwal** [comment, marked as answer]
Yes, GitHub's API rate limit can sometimes be a challenge when developing applications that interact with the API frequently. However, as the error message suggests, authenticated requests get a higher rate limit. Here are some steps you can take to increase your API rate limit:

1. Authenticate your requests: GitHub offers a higher rate limit for authenticated requests. You can authenticate your requests using an access token or OAuth token. You can create a personal access token on GitHub and use it in your application.

2. Use pagination: If you are retrieving large amounts of data, consider using pagination to retrieve the data in smaller chunks. This can help reduce the number of requests you make and reduce the likelihood of hitting the rate limit.

3. Cache responses: If you are making the same requests repeatedly, consider caching the responses to reduce the number of requests you make. This can help reduce the load on the API and improve the performance of your application.

4. Contact GitHub support: If you are still hitting the rate limit even after trying the above steps, you can contact GitHub support to request a higher rate limit. However, this is usually only granted for specific use cases and is not guaranteed.

By following these steps, you should be able to reduce the likelihood of hitting the API rate limit and improve the performance of your application.

**Reply by scffs** [reply]
Okey, thank you

**Reply by scffs** [reply]
So, I'm sorry, but how can I ask GitHub support about a higher rate limit?
I didn't find anything at https://support.github.com/

**Reply by Anmol-Baranwal** [reply]
You should check [here](https://support.github.com/request?tags=docs-generic)
I haven't contacted GitHub support yet, so I doesn't really know the process.
